### ISSUE TITLE
LineCanvas doesn't allow a one row/col join

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#2433]
I'm trying to use LineCanvas to do the automatic thing for the new border/title style @tznind has demoed in the new FileDialog:

```
╔╡ Title ╞═════╗
```

This code SHOULD do it:

```cs
// ╔╡ Title ╞═════╗
// Add a short horiz line for ╔╡
lc.AddLine (screenBounds.Location, 1, Orientation.Horizontal, BorderStyle);
// Add a short vert line for ╔╡
lc.AddLine (new Point (screenBounds.X + 1, screenBounds.Location.Y - 1), 1, Orientation.Vertical, BorderStyle.Single);
// Add a short vert line for ╞
lc.AddLine (new Point (screenBounds.X + (Parent.Title.Length + 3), screenBounds.Location.Y - 1), 1, Orientation.Vertical, BorderStyle.Single);
// Add the right hand line for ╞═════╗
lc.AddLine (new Point (screenBounds.X + (Parent.Title.Length + 3), screenBounds.Location.Y), Frame.Width - 1 - (Parent.Title.Length + 2), Orientation.Horizontal, BorderStyle);
```

But it doesn't:

<img width="262" alt="image" src="https://user-images.githubusercontent.com/585482/226253544-77d3acf4-99f8-4338-83b2-5a7b6c13aa7b.png">

If I try `screenBounds.Y - 1` and a length of `1` I get this, which is close...

<img width="260" alt="image" src="https://user-images.githubusercontent.com/585482/226253793-30833f9b-bb72-46c4-850e-08ef318a822b.png">

And with a length of `2`: 

<img width="295" alt="image" src="https://user-images.githubusercontent.com/585482/226253413-939f883c-2a49-49d6-947f-9def59b0b2e0.png">

**Comment by tznind** [comment]
Will have a think about this but am short on time atm.  Here is a unit test that does mostly what you want I think:

```csharp
[Fact, AutoInitShutdown]
public void TestLineCanvas_ClipArea_Intersections ()
{
	// Draw at 1,1 within client area of View (i.e. leave a top and left margin of 1)
	var v = GetCanvas (out var lc);
	v.Width = 10;
	v.Height = 1;
	v.Bounds = new Rect (0, 0, 10, 1);

	// ╔╡ Title ╞═════╗
	// Add a short horiz line for ╔╡
	lc.AddLine (new Point(0,0), 1, Orientation.Horizontal, BorderStyle.Double);
	//LHS line down
	lc.AddLine (new Point (0, 0), 5, Orientation.Vertical, BorderStyle.Double);

	//Vertical line before Title (must cover 3 squares so it results in a ╡ intersection
	lc.AddLine (new Point (1, -1), 3, Orientation.Vertical, BorderStyle.Single);
	//Vertical line after Title (must cover 3 squares so it results in a ╞ intersection
	lc.AddLine (new Point (6, -1), 3, Orientation.Vertical, BorderStyle.Single);
	
	// remainder of title
	lc.AddLine (new Point (6, 0), 3, Orientation.Horizontal, BorderStyle.Double);
	//RHS line down
	lc.AddLine (new Point (9, 0), 5, Orientation.Vertical, BorderStyle.Double);

	v.Redraw (v.Bounds);

	string looksLike =
@"
╔╡    ╞══╗
";
	TestHelpers.AssertDriverContentsAre (looksLike, output);
}
```

**Comment by tznind** [comment]
Probably tied up in this is that having a length of 0 gives you a dot `.` and having a length of 1 gives you a line from point a to point b where point b is 1 unit away:

![image](https://user-images.githubusercontent.com/31306100/226291613-421e09b4-d61c-48ce-aa39-a12b08d5ea60.png)

This gives you 2 character width line on rendering (see `TestLineCanvas_Horizontal`):
`──` 

This is more correct than it sounds because if you have corners then you end up with:
`┌┐`

Notice how it looks half the length of the one above when really it isn't (its still 2 characters wide).

**Comment by tig** [comment]
Fixed by #2436
